pallets/validators-manager/src/lib.rs

```diff
@@ -619,6 +630,58 @@ impl<T: Config> Pallet<T> {
         Ok(hex::encode(EthAbiHelper::generate_eth_abi_encoding_for_params_only(&eth_description)))
     }
 
+    /// This function generates the data needed to generate a confirmation for registering a new
+    /// collator. The implementation must match this schema:
+    /// https://github.com/Aventus-Network-Services/avn-bridge/blob/v1.1.0/contracts/AVNBridge.sol#L344-L345
+    pub fn convert_collator_registration_action_to_confirmation_signing_data(
+        action_id: &ActionId<T::AccountId>,
+        enable_data_logs: bool,
+    ) -> Result<String, DispatchError> {
+        let validators_action_data = Self::try_get_validators_action_data(action_id)?;
+        let mut activate_collator_params_concat: [u8; PACKED_ARGUMENTS_SIZE] = [0u8; 96];
+
+        let activate_collator_params = match validators_action_data.reserved_eth_transaction {
+            EthTransactionType::ActivateCollator(ref d) => d,
+            _ => Err(Error::<T>::ErrorGeneratingEthDescription)?,
+        };
+
+        activate_collator_params_concat[0..64]
+            .copy_from_slice(&activate_collator_params.t1_public_key.as_bytes()[0..64]);
+        activate_collator_params_concat[64..96]
+            .copy_from_slice(&activate_collator_params.t2_public_key[0..32]);
+
+        let activate_collator_hash = keccak_256(&activate_collator_params_concat);
+
+        let sender =
+            T::AccountToBytesConvert::into_bytes(&validators_action_data.primary_validator);
+
+        // Now treat this as an bytes32 parameter and generate signing abi.
+        let hex_encoded_confirmation_data =
+            hex::encode(EthAbiHelper::generate_confirmation_data_for_compacted_calls(
+                &activate_collator_hash,
+                validators_action_data.eth_transaction_id,
+                &sender,
+            ));
+
+        if enable_data_logs {
+            log::info!(
```

Comment:
we can also move this into the separate function

Reply:
I don't think using a function will improve readability.

Reply:
I think it would significantly improve the readability

Reply:
If the function is well named, it should. It always should.